tests/data/test_data/example_netcdf_generator.py

```diff
@@ -0,0 +1,193 @@
+"""Generate NetCDF datasets for testing."""
+
+
+def generate_files() -> None:
```

Comment:
I know this is here to generate a bunch of netCDF files for testing on the fly, but you really should be following the same practices than in any other part of the code - in particular having each function to do one thing only. This one is creating a collection of files with different contents and properties. 

I suggest to break this into as many functions as files you are generating.

Reply:
I definitely see what you mean, but honestly the single function of this file is always to generate the complete set of files. The files are generated and added to the repo, not run during testing or anything, so I can't see any circumstances when those 14 separate file creations would need to be isolated from one another. The `if __name__ == "__main__":` line would be followed by 14 separate function calls.

We _could_ expand the script to allow users to re-generate particular files or add new ones, but honestly it is so quick to run and so infrequently run, that the overhead of maintaining a CLI to do that seems over engineered.

Reply:
The second part it is definitely over-engineering the problem. About the first one... I think it is just a question of consistency.

Having said that, it is definitely not a big deal and should not delay this PR!